### PR TITLE
Kraken : stop_point management in traffic_reports

### PR DESCRIPTION
**Pull request by pbougue**
also : bss doc (unrelated but small)

### CODE REVIEW

File: documentation/slate/source/includes/apis.md [hunk 1]
```diff
@@ -958,6 +958,7 @@ The [isochrones](#isochrones) service exposes another response structure, which
 | nop     | disruption_active    | boolean | For compatibility use only.<br>If true the algorithm take the disruptions into account, and thus avoid disrupted public transport.<br>Rq: `disruption_active=true` = `data_freshness=realtime` <br>Use `data_freshness` parameter instead       |  False     |
 | nop     | wheelchair           | boolean | If true the traveler is considered to be using a wheelchair, thus only accessible public transport are used<br>be warned: many data are currently too faint to provide acceptable answers with this parameter on       | False       |
 | nop     | direct_path          | enum    | Specify if direct paths should be suggested.<br>possible values: <ul><li>indifferent</li><li>none</li><li>only</li></ul>      | indifferent |
+| nop     | bss_stands           | boolean | Activate the output of BSS availability in the bicycle_rental pois of response     | False       |
```
Comment (TeXitoi) [inline documentation/slate/source/includes/apis.md:962]: Please add this documentation to the other API using this flag (places_nearby at least).

> :white_check_mark: Done only on places_nearby, not available in places (also on places_uri and poi, so didn't mention it)

File: source/disruption/traffic_reports_api.cpp [hunk 2]
```diff
@@ -121,50 +122,61 @@ NetworkDisrupt& TrafficReport::find_or_create(const type::Network* network){
...
+           LOG4CPLUS_WARN(logger, "Disruption::add_stop_points : Unable to parse filter "
+                               + parse_error.more);
+       } catch (const ptref::ptref_error& /*ptref_error*/) {
+          // that can arrive quite often if there is a filter, and
+          // it's quite normal. Imagine /line/metro1/traffic_reports
+          // for the network SNCF.
+       }
+
+       // build a map of messages per stop_area (iterate only on stop_points of the network)
+       std::map<const nt::StopArea*, std::vector<boost::shared_ptr<nt::disruption::Impact>>> sa_messages;
```
Comment (TeXitoi) [inline source/disruption/traffic_reports_api.cpp:150]: Maybe you need a `std::map<const nt::StopArea*, std::set<boost::shared_ptr<nt::disruption::Impact>>>`

File: source/disruption/traffic_reports_api.cpp [hunk 3]
```diff
@@ -121,50 +122,61 @@ NetworkDisrupt& TrafficReport::find_or_create(const type::Network* network){
...
+               // add stop_area messages only once
+               sa_messages[sa] = sa->get_publishable_messages(now);
+           }
+           // add stop_point messages
+           auto sp_mess = sp->get_publishable_messages(now);
+           sa_messages[sa].insert(sa_messages[sa].end(), sp_mess.begin(), sp_mess.end());
+       }
+
+       for (auto sa_mess : sa_messages) {
+           if (!sa_mess.second.empty()) {
```
Comment (TeXitoi) [inline source/disruption/traffic_reports_api.cpp]: continue

File: source/disruption/traffic_reports_api.cpp (hunk 2 above)
Comment (pbougue) [reply source/disruption/traffic_reports_api.cpp:150]: Doesn't look like it's needed, so added tests for that: https://github.com/CanalTP/navitia/pull/2146/files#diff-ca942c5646d0c983ab93aceac6cccb19R120

**Review by TeXitoi — APPROVED** [review]

File: source/disruption/traffic_reports_api.cpp [hunk 4]
```diff
@@ -121,50 +122,62 @@ NetworkDisrupt& TrafficReport::find_or_create(const type::Network* network){
...
+           if (sa_messages.find(sa) == sa_messages.end()) {
+               // add stop_area messages only once
+               sa_messages[sa] = sa->get_publishable_messages(now);
+           }
+           // add stop_point messages
+           auto sp_mess = sp->get_publishable_messages(now);
+           sa_messages[sa].insert(sa_messages[sa].end(), sp_mess.begin(), sp_mess.end());
+       }
+
+       for (auto sa_mess : sa_messages) {
```
Comment (antoine-de) [inline source/disruption/traffic_reports_api.cpp]: const&

**Review by antoine-de — APPROVED** [review]

**Pull request merged**